ui/tests/test_learningresources_views.py

```diff
@@ -84,15 +84,23 @@ def test_create_repo_post(self):
         body = resp.content.decode("utf-8")
         self.assertTrue(repo_name in body)
 
-    def test_listing_unauthorized(self):
-        """View listing page."""
-        # Not authorized to view this repository...
-        body = self.assert_status_code(
+    def test_listing_not_found(self):
+        """View listing page, but repo does not exist."""
+        self.assert_status_code(
             "/repositories/99/",
+            NOT_FOUND,
+            return_body=True
+        )
+
+    def test_listing_unauthorized(self):
+        """View listing page, but not authorized to view this repository."""
+        self.logout()
+        self.login(self.USERNAME_NO_REPO)
+        self.assert_status_code(
+            self.repository_url,
             UNAUTHORIZED,
             return_body=True
         )
-        self.assertTrue("unauthorized" in body)
```

Comment:
The body does not contain the string any more because now I raise an exception instead of returning a response object